src/cli_chess/core/game/offline_game/offline_game_model.py

```diff
@@ -49,12 +50,44 @@ def make_move(self, move: str):
 
                 if not self.is_my_turn():
                     raise Warning("Not your turn")
-
                 move = move.strip()
                 if not move:
                     raise Warning("No move specified")
-
+                # clean premove
+                self.board_model.set_premove(None)
                 self.board_model.make_move(move)
+
+            except Exception:
+                raise
+        else:
+            log.warning("Attempted to make a move in a game that's not in progress")
+            raise Warning("Game has already ended")
+        
+    def make_premove(self, move: str):
+        """Verify move and Set board model premove"""
+        if self.game_in_progress:
+            try:
+                if self.board_model.board.is_game_over():
+                    self.game_in_progress = False
+                    raise Warning("Game has already ended")
+                if self.board_model.get_premove() is not None:
+                    raise Warning("You already have a premove set")
+                if self.is_my_turn():
+                    raise Warning("It is your turn, cant not make a premove")
```

Comment:
Remove as this should not be hit.
```suggestion
                if self.is_my_turn():
                    raise Warning("It is your turn, cant not make a premove")
```